packages/cli/src/ceramic-cli-utils.ts

```diff
@@ -414,38 +414,44 @@ export class CeramicCliUtils {
 
   /**
    * Pin stream
+   * @param privateKey - Admin DID private key
    * @param streamId - Stream ID
    */
-  static async pinAdd(streamId: string): Promise<void> {
+  static async pinAdd(privateKey: string, streamId: string): Promise<void> {
     const id = StreamID.fromString(streamId)
 
-    await CeramicCliUtils._runWithCeramicClient(async (ceramic: CeramicApi) => {
+    await CeramicCliUtils._runWithCeramicClient(async (ceramic: CeramicClient) => {
+      await CeramicCliUtils._authenticateClient(ceramic, privateKey)
```

Comment:
`CeramicCliUtils._runWithCeramicClient` already have a DID: https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/cli/src/ceramic-cli-utils.ts#L501

Reply:
err, here: https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/cli/src/ceramic-cli-utils.ts#L499

Reply:
Imo already confusing enough that the CLI and the DID used for CAS Auth have different ways to provide seeds. No need to add yet another way to create a DID when running js-ceramic.

Reply:
Isn't that different? My understanding is that the CLI stores a seed locally, but that doesn't mean it's necessarily an admin DID?
If an operator wants to interact with pins on a remote server, they should be able to provide the admin DID private key from the client, no?

Reply:
> If an operator wants to interact with pins on a remote server, they should be able to provide the admin DID private key from the client, no?

yes, they should configure this for their cli client with the right seed to do this.